lab_4_fill_words_by_ngrams/main.py

```diff
@@ -220,6 +319,29 @@ def _calculate_perplexity(self, generated_text: str) -> float:  # type: ignore
                 or if methods used return None,
                 or if nothing was generated.
         """
+        if not generated_text:
+            raise ValueError('Input argument is empty')
+        if not isinstance(generated_text, str):
+            raise ValueError('Inappropriate type argument')
+        encoded_text = self._word_processor.encode(generated_text)
+        if encoded_text is None:
+            raise ValueError('self._word_processor.encode() returned None')
+        self._language_model.build()
```

Comment:
is this the responsibility of quality checker?

Reply:
fixed